factor/iprange/classes/factor.php

```diff
@@ -31,4 +31,73 @@
 
 class factor extends object_factor_base {
 
+    /**
+     * IP Range Factor implementation.
+     * This factor needs no user setup, return true.
+     *
+     * {@inheritDoc}
+     */
+    public function setup_user_factor($data) {
+        return true;
+    }
+
+    /**
+     * IP Range Factor implementation.
+     * This factor is a singleton, return single instance.
+     *
+     * {@inheritDoc}
+     */
+    public function get_all_user_factors() {
+        $factor = (object) array(
+            'id' => 1,
+            'name' => $this->name,
+            'devicename' => '-',
+            'timecreated' => '-',
+            'createdfromip' => '-',
+            'lastverified' => '-',
```

Comment:
same as auth, it would be good to make this a real record and store when we last verified this account was on the secure network

Reply:
I'll merge regardless, we can circle back to this when we add the reporting